faucet-api/Controllers/FaucetController.cs

```diff
@@ -70,7 +73,30 @@ public async Task<IActionResult> SendFunds(string address, long? amount)
         {
             return await SendFunds(new SendRequest {ToAddress = address, Amount = amount ?? 20});
         }
+        private async Task RefillUtxoPool(BitcoinAddress fromAddress)
+        {
+            lock (_lockObject) {
+                if (_utxoPool.Count > _poolThreshold) return;
+            }
+            var utxos = (await _indexerService.FetchUtxoAsync(fromAddress.ToString(), 0, 50)) ?? [];
+            lock (_lockObject)
+            {
+                _utxoUsed.IntersectWith(utxos);
```

Comment:
why call  `_utxoUsed.IntersectWith(utxos)` ?  
seems to me we don't need to remove anything that comes from the indexer anyway the next line will do that 
`_utxoPool.UnionWith(utxos.Except(_utxoUsed))`
here we will anyway remove anything that is in use, maybe I did not think of a scenario